docs/website/contents/for-developers/PreflightGuide.md

```diff
@@ -128,6 +128,124 @@ Use a ledger snapshot of your local node (make sure to add the `_db-analyser` su
+The exact probabilities depend on the stake distribution.
+In general, the more equally the stake is distributed across many pools, the more likely are slots with more slot leaders.
+In the extreme case, which one can think of as the stake $\sigma$ being distributed evenly across infinitely many pools, the number of slot leaders $L$ of a single slot is given by a Poisson distribution $\mathop{\mathrm{Pois}}(-\sigma\log(1-f))$.
+
+For example, consider $\sigma = 1$:
+
+![](./PreflightGuide/plot-single-slot-leaders.png)
+
+We see that having three or more leaders per slot quickly becomes very (exponentially) unlikely.
+
+In particular, we can calculate the probability that a slot is an active multi leader slot
+```math
+\mathop{\mathrm{Pr}}(L \ge 2)
+  = 1-\mathop{\mathrm{Pr}}(L = 0)-\mathop{\mathrm{Pr}}(L=1)
+  = f + \sigma \cdot {(1-f)}^\sigma \cdot \log(1-f)
+```
+via the definition of the Poisson distribution and $L \sim \mathop{\mathrm{Pois}}(-\sigma\log(1-f))$.
+For $f=1/20$ and $\sigma =1$, we get $\mathop{\mathrm{Pr}}(L \ge 2) = 0.127\%$, ie 0.127% of all slots are multi leader slots in the worst case.
+In that case, we would expect to see $10k/f \cdot 0.00127 \approx 549$ multi leader slots per epoch.
+
+On the community-maintained [pooltool.io](https://pooltool.io/networkhealth), we can observe the number of "slot battles" per epoch, ie the number of times the nodes reporting to pooltool.io observed two blocks in the same slot.
+
+![](./PreflightGuide/plot-pooltool-slot-battles.png)
+
+As the stake on mainnet is "only" distributed across a few thousand pools (and not completely evenly), we are not in the extreme case modeled by the Poisson described above.
+And indeed, we usually see even less than 500 slot battles per epoch.
+
+### Grinding
+
+The discussion above completely ignored *grinding*.
+In the context of the leader schedule, this refers to an adversarial influence on the nonce that is used as a seed for all of the VRF computations for slot election within a specific epoch.
+The exact mechanism is out-of-scope for this section, but it should be said that such an attack requires both a considerable amount of stake and computational resources, and it allows an adversary to choose the best out of multiple epoch nonces.
+An extremely conservative upper bound for how many nonces an attacker can choose from is $10^{20}$ (based on the computational resources required to even compute all of these nonces).
+The mainnet parameters of Cardano (in particular the security parameter) were chosen with resistance against grinding in mind.
+
+Concretely, an attacker with $\alpha$ stake that can choose out of $N$ epoch nonces can ensure that it leads any specific slot with probability $1-{(1-\phi(\alpha))}^N$, which is a huge advantage compared to $\phi(\alpha)$.
+For example, an attacker with $N=10^{20}$ with even a very small amount of stake has a chance of close to 100% to cause themselves to be elected in any specific slot.
```

Comment:
I understand this as "An attacker can choose (with close to 100% probability) a slot in which they want to be elected" but this doesn't mean that they can choose with 100% probability to be elected in say 5 specific slots in the epoch right? The 100% applies if a single slot is targeted no?

Reply:
It probably also still applies to 5 slots, but not to large numbers of slots, the next paragraph is also supposed to point this out (slightly modified in the latest commit)

> Usually, an attacker wants to maximize their total number of active slots within a larger number of slots, like in an entire epoch. For this, the impact of grinding is smaller (but still large), as every epoch nonce affects the entire leader schedule of the epoch, and the attacker can not choose different epoch nonces for different slots.

Does that make sense, or do we want to make this more explicit?

Reply:
I think just modifying this word would be enough:
```suggestion
For example, an attacker with $N=10^{20}$ with even a very small amount of stake has a chance of close to 100% to cause themselves to be elected in a specific slot.
```

```suggestion
For example, an attacker with $N=10^{20}$ with even a very small amount of stake has a chance of close to 100% to cause themselves to be elected in any single specific slot.
```

Reply:
But also feel free to ignore this comment :smile: